tutorials/hardhat-tutorial.md

```diff
@@ -0,0 +1,425 @@
+      // objects have a lot of utility methods to assert values.
+      expect(await deployedDao.target()).to.equal(targetCollator);
+});
+```
+
+Let's add another test case. When a StakingDAO is launched, it shouldn't have any funds. This test verifies that is indeed the case. Go ahead and add the following test case to your `Dao.js` file:
+
+```javascript
+it("The DAO should initially have 0 funds in it", async function () {
+	const { deployedDao, deployer } = await loadFixture(deployDaoFixture);
+	expect(await deployedDao.totalStake()).to.equal(0);
+});
+```
+
+### Function Reverts {: #function-reverts }
+
+To this point the two cases implemented have been simple but important. Now, you'll implement a more complex test case that differs in its architecture. In prior examples you've verified that a function returns an expected value. In this one, you'll be verifying that a function reverts. You'll also change the address of the caller to test an admin-only function. 
+
+In the [StakingDAO contract](https://github.com/PureStake/moonbeam-intro-course-resources/blob/main/delegation-dao-lesson-one/DelegationDAO.sol){target=_blank}, only admins are authorized to add new members to the DAO. One could write a test that checks to see if the admin is authorized to add new members but perhaps a more important test is to ensure that *non-admins* can't add new members. To run this test case under a different account, you're going to ask for another address when you call `ethers.getSigners()` and specify the caller in the assertion with `connect(otherAddress)`. Finally, after the function call to be tested you'll append `.to.be.reverted;` to indicate that the test case is successful if the function reverts. And if it doesn't revert it's a failed test! 
+
+```javascript
+it("Non-admins should not be able to grant membership", async function () {
+	const { deployedDao, deployer } = await loadFixture(deployDaoFixture);
+	const [account1, otherAddress] = await ethers.getSigners();
+	await expect(deployedDao.connect(otherAddress).grant_member("0x0000000000000000000000000000000000000000")).to.be.reverted;
+});
+```
+
+### Swapping Fixtures {: #swapping-fixtures }
+
+For this example, you'll load another fixture - the one that has a DAO deployed with a member. Then, you'll check to verify whether that member can call the `check_free_balance()` function of StakingDAO, which has an access modifier such that only members can access it. 
+
+```javascript
+it("DAO members should be able to access member only functions", async function () {
+	const { deployedDao, deployer } = await loadFixture(deployDaoFixtureWithMembers);
+    const [account1, member1] = await ethers.getSigners();
+    expect(await deployedDao.connect(member1).check_free_balance()).to.equal(0);
+	});
+  });
+});
+```
+And that's it! You're now ready to run your tests! Since this is the last test case completing the test file, a series of closing brackets are added to the prior code snippet.  
```

Comment:
ah ok I see now you mentioned adding in the closing brackets, but I feel like it's better to have them in each of the snippets in case a user wants to run the tests as they add them 🤷‍♀️